### PR TITLE
Add MSVC 14.37.

### PR DESCRIPTION
**Pull request by wmaxey**
This bumps MSVC 2022 to include 14.37 and drops 14.34.

### CODE REVIEW

**Review by miscco — APPROVED** [review]

**Comment by trxcllnt** [comment]
@wmaxey https://github.com/rapidsai/devcontainers/actions/runs/6801523024/job/18492678369?pr=171#step:4:22

**Comment by wmaxey** [comment]
> @wmaxey https://github.com/rapidsai/devcontainers/actions/runs/6801523024/job/18492678369?pr=171#step:4:22

Pushed a fix.

**Comment by wmaxey** [comment]
@miscco Can you send a `/ok to test`?

File: matrix.yml [hunk 1]
```diff
@@ -103,9 +103,9 @@ include:
   - {features: [{<<: *cuda_curr_max, <<: *cccl_cuda_opts}, { name: "cl", version: "14.27" }]} # MSVC 2019
   - {features: [{<<: *cuda_curr_max, <<: *cccl_cuda_opts}, { name: "cl", version: "14.28" }]} # MSVC 2019
   - {features: [{<<: *cuda_curr_max, <<: *cccl_cuda_opts}, { name: "cl", version: "14.29" }]} # MSVC 2019
-  - {features: [{<<: *cuda_curr_max, <<: *cccl_cuda_opts}, { name: "cl", version: "14.34" }]} # MSVC 2022
```
Comment (miscco) [inline matrix.yml:106]: Why is this removing 19.34 here but leaving it in below?

**Review by miscco — CHANGES_REQUESTED** [review]

File: windows/vs-version-matrix.ps1 [hunk 2]
```diff
@@ -1,7 +1,7 @@
 $global:vsVerToCompilers = @{
     "2017"=("14.14", "14.15", "14.16");
     "2019"=("14.27", "14.28", "14.29");
-    "2022"=("14.34", "14.35", "14.36");
+    "2022"=("14.35", "14.36", "14.36");
```
Comment (wmaxey) [inline windows/vs-version-matrix.ps1:4]:
```suggestion
    "2022"=("14.35", "14.36", "14.37");
```

File: matrix.yml (hunk 1 above)
Comment (wmaxey) [reply matrix.yml:106]: Like in the rest of the files? Deleting them from the other files would prevent us from building it locally... If we needed to.

I'm fine with removal.

Comment (miscco) [reply matrix.yml:106]: I would just leave it in. We do have most other versions still

**Pull request merged**